lib/replicate.js

```diff
@@ -345,11 +317,11 @@ Peer.prototype.oncancel = function () {
 }
 
 Peer.prototype.onclose = function () {
-  this.destroy()
+  this._close()
 }
 
 Peer.prototype.have = function (have) { // called by feed
-  if (this.stream && this.remoteWant) this.stream.have(have)
+  if (this.stream && this.remoteWant && this._remoteOpened) this.stream.have(have)
```

Comment:
When would `this.remoteWant` be true when `this._remoteOpened` isn't? I'd imagine that all wants would have to be received after the first open right?

Reply:
good point!